turbopack/crates/turbo-rcstr/src/dynamic.rs

```diff
@@ -172,24 +172,20 @@ const fn multiply_mix(x: u64, y: u64) -> u64 {
     }
 }
 
-// Const compatible helper function to read a u64 from a byte array at a given offset
+// Const compatible helper function to read a u64 from a byte array at a given
+// offset
+#[inline(always)]
 const fn read_u64_le(bytes: &[u8], offset: usize) -> u64 {
-    (bytes[offset] as u64)
-        | ((bytes[offset + 1] as u64) << 8)
-        | ((bytes[offset + 2] as u64) << 16)
-        | ((bytes[offset + 3] as u64) << 24)
-        | ((bytes[offset + 4] as u64) << 32)
-        | ((bytes[offset + 5] as u64) << 40)
-        | ((bytes[offset + 6] as u64) << 48)
-        | ((bytes[offset + 7] as u64) << 56)
+    let array = unsafe { bytes.as_ptr().add(offset) } as *const [u8; 8];
+    u64::from_le_bytes(unsafe { *array })
```

Comment:
Can you add some sort of compile-time assertion, maybe using `cfg` that the architecture is LE? In practice, every platform we support is LE, so it's okay to just fail to compile if we're on a BE architecture, but I'd like some explicit guarantee of that.

Reply:
will clarify this with a comment